docs/create_new_activity_protocol.md

```diff
@@ -0,0 +1,122 @@
+
+Creating `activityName_schema` – use the keys defined in [`schemas/Activity`](./schemas/Activity). If any other keys are used, then define them in `activityName_context`
+
+For example,
+
+```json
+  {
+  "@context": [ "https://raw.githubusercontent.com/ReproNim/reproschema/master/contexts/generic",
+      "https://raw.githubusercontent.com/ReproNim/reproschema/master/activities/Wellbeing/Wellbeing_context"
+  ],
+  "@type": "reproschema:Activity",
+  "@id": "Wellbeing_schema",
+  "skos:prefLabel": "Wellbeing",
+  "schema:description": " Wellbeing Voice tasks",
+  "schema:schemaVersion": "0.0.1",
+  "schema:version": "0.0.1",
+  "preamble": {
+      "en": "For each task, you should hit the record button before speaking and then stop once you are done speaking. You may hit play to hear what was recorded.",
+      "es": "Para cada pregunta, debería hacer click sobre el botón antes de hablar y luego hacerlo de nuevo para parar de grabar. Luego puede tocar play para escuchar su respuesta."
+  },
+  "ui": {
+      "addProperties": [
+          {"isAbout": "share_data",
+          "variableName": "share_data",
+          "prefLabel": {"en": "Share Data"},
+          "isVis": true,
+          "allow": ["skipped", "dontknow"],
+          ...
+          },
+          {"isAbout": "email",
+          "variableName": "email",
+          "isVis": "share_data === 1"
+          },
+          {"isAbout": "multipart_audio_check",
+          "variableName": "multipart_audio_check",
+          "isVis": true
+          },
+          {"isAbout": "free_speech_general_mood",
+          "variableName": "share_data",
+          "isVis": true
+          },
+          {"isAbout": "say_ah",
+          "variableName": "say_ah",
+          "isVis": true
+          }
+      ],
+      "order": [
+          "multipart_audio_check",
+          "share_data",
+          "email",
+          "free_speech_general_mood",
+          "say_ah"
+      ],
+      "shuffle": false
+  }
+}
+```
+
+Mandatory keys:
+
+- `@context` - [Array] Include the ReproNim generic context JSON-LD file along with the activity context.
+- `@type`- describes type of the schema.
+- `@id` - unique id for the schema. should be same as the filename.
+- `skos:prefLabel` - display name for the schema
+- `ui.addProperties` - defines the various properties of each item.
+- `variablename` - variable name used in `ui.order` for the items
+- `isAbout` - file name of the corresponding variable name
+- `ui.order` - [Array] defines the order in which the items appear in the activity
+
+- To create `item_x` in the items folder:
+
+  - Use keys defined in [`schemas/Field`](./schemas/Field)
+  - `@type`=`"https://raw.githubusercontent.com/ReproNim/reproschema/master/schemas/Field"`
+  - `responseOptions` – can be embedded or can point to a remote JSON-LD object.
+
+### To create a protocol
+
+  - Under the [`protocols`](./protocols) directory, create directory with name of protocol in the CamelCase naming convention.
```

Comment:
@sanuann it seems that the CamelCase is not strictly enforced, no?

We currently have 3 protocols that use PascalCase and the others are more like "no case" :wink : 
- Malargue
- TokenLogger
- VoicePilot
- wellbeing
- mindlogger-demo

The same goes for the activities: I see mostly a mix of snake_case and PascalCase.

Fine by me but it is weird to put in the doc a rule we do not respect. :smile:

Reply:
that is an outdated rule. 😄 
perhaps we should just mention it as a recommendation. what say?

Reply:
OK will adapt. :-)